### PR TITLE
Try faster_build

### PR DESCRIPTION
**Pull request by evgenytalanin-intel**

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
@evgenytalanin-intel can we enable this? Looks like it works well for Linux, Windows at least

CC @vinograd47

File: cmake/features.cmake [hunk 1]
```diff
@@ -49,4 +49,4 @@ ie_dependent_option (ENABLE_PROFILING_ITT "ITT tracing of IE and plugins interna
 # Documentation build
 ie_option (ENABLE_DOCS "build docs using Doxygen" OFF)
 
-ie_dependent_option (ENABLE_FASTER_BUILD "Enable build features (PCH, UNITY) to speed up build time" OFF "CMAKE_VERSION VERSION_GREATER_EQUAL 3.16" OFF)
+ie_option (ENABLE_FASTER_BUILD "Enable build features (PCH, UNITY) to speed up build time" ON)
```
Comment (vinograd47) [inline cmake/features.cmake:52]: Minimal supported CMake version is `3.13`, so we have to check the actual CMake version.

Comment (ilya-lavrenov) [reply cmake/features.cmake:52]: we have minimum required version 3.13 in root cmake.

**Comment by ilyachur** [comment]
As I remember we made some code changed to enable UNITY.

If we enable this in CI, I think we need to update a contribution guide in order to user has an information how he can reproduce the issue if he broke something in CI.

**Comment by ilya-lavrenov** [comment]
Faster build is enabled on public CI https://github.com/openvinotoolkit/openvino/pull/3708

File: cmake/features.cmake (hunk 1 above)
Comment (vinograd47) [reply cmake/features.cmake:52]: Yes, generic requirement is `3.13`, but Unity and PCH requires `3.16`. Either we need to change base version check or leave this option as dependent.

**Pull request closed without merging**